tensorrt_llm/_torch/pyexecutor/_util.py

```diff
@@ -175,6 +176,20 @@ def try_prepare_estimation(self) -> bool:
             estimating_kv_cache = True
             self._executor_config.kv_cache_config.max_tokens = self._get_token_num_for_estimation(
             )
+            maw = self._executor_config.kv_cache_config.max_attention_window
+            if maw is not None:
+                unique_windows = list(set(maw))
+                if len(unique_windows) > 1:
+                    # NOTE: For sliding window, we need to allocate more tokens(max_num_tokens + window_size) for each window, need to update if this requirement changes
+                    assert len(
+                        unique_windows
+                    ) == 2, "Sliding window with more than 2 window sizes has not been tested"
+                    # sliding window is the one that is not the executor_config.max_seq_len
+                    sliding_window = unique_windows[0] if unique_windows[
+                        0] != self._executor_config.max_seq_len else unique_windows[
+                            1]
+                    self._executor_config.kv_cache_config.max_tokens += (
```

Comment:
I don't understand this?
The result will be:
`self._get_token_num_for_estimation() + self._executor_config.max_num_tokens + sliding_window`.
What does this calulcation correspond to?